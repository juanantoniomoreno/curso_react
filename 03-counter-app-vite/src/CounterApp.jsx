import { useState } from 'react'
import PropTypes from 'prop-types'
import { Button } from 'react-bootstrap'

export const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState( value );
  
  const handleAdd = () => { 
    
    setCounter( counter + 1 );
    //setCounter( (c) => c + 1  )
  }

  const handleSubstract = () => {
    setCounter( counter - 1 );
  }

  const handleReset = () => {
    setCounter( value );
  }

  return (
    <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>
        <div>
          <Button 
            variant='info'
            onClick={ handleAdd }
            >+1
          </Button>
          <Button 
            variant='warning'
            onClick={ handleSubstract }
            > -1
          </Button>
          <Button 
            variant='secondary'
            onClick={ handleReset }
            aria-label = "btn-reset"
            >Reset
          </Button>
        </div>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}


