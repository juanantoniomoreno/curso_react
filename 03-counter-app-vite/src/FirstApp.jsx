import PropTypes from 'prop-types';

const getSaludo = (name) => {
  return `Hola ${name}`;
}

const getAge = (birth_year) => {
  let now = new Date();
  return (now.getFullYear() - birth_year);
}

export const FirstApp = ({ title, subTitle, name }) => {

  //console.log(props);
      
  return (
    <>
      <h1 data-testid="test-title" > { title }</h1>
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      {/* <h3>Tengo { subTitle } años</h3> */}
      {/* <code>{ JSON.stringify( newMsg ) }</code> */}
      <p>{ name }</p>
    </>
  )
}

//Para hacer que las props sean obligatorias y sean de un tipo determinado
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
}

//Podemos dar valores por defecto a las props
FirstApp.defaultProps = {
  /* title: 'No hay título', */
  subTitle: 'No hay subtítulo',
  name: 'Juan'
}
