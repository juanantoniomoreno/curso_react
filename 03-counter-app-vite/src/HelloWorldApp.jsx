import React from 'react'
import { FirstComponent } from './FirstComponent'

export const HelloWorldApp = () => {
  return (
    <div>
      Hola Mundo!!
    </div>
  )
}