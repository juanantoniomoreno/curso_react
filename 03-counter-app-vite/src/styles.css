html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 50px;
}

button {
    margin: 0.1rem;
}