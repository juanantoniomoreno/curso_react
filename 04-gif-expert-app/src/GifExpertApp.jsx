import { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = () => {

        setCategories([ 'Valorant', ...categories ])
    }

    console.log( categories );
    
    

  return (
    <>
        {/* *TITULO */}
        <h1>GifExpertApp</h1>

        {/* INPUT */}
        <AddCategory />

        {/* LISTADO DE GIFT */}
        <button onClick={ onAddCategory }>Agregar</button>
        <ol>
            {categories?.map( (category, index) => {
                return(
                    <li key={ index }>{category}</li>
                )
            })}
        </ol>

    </>
  )
}
