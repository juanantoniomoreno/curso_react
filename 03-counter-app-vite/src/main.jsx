import React from 'react'
import ReactDOM from 'react-dom/client';
import 'bootstrap/dist/css/bootstrap.min.css';
import { CounterApp } from './CounterApp';
/* import { FirstApp } from './FirstApp'; */

import './styles.css';

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title='Hola, Soy Vegeta'/> */}
        <CounterApp value={ 0 }/>
    </React.StrictMode>
);