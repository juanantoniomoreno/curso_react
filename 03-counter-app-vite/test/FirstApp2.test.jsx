import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp/>', () => {

    const title = 'Hola soy Goku';
    const subTitle = 'soy  un subtitulo';

    test('Debe hacer match con el snapshot', () => {

        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();
    });

    test('debe mostrar el mensaje "Hola, Soy Goku', () => {
        
        render( <FirstApp title={title} /> );
        //screen.debug();

        expect( screen.getByText(title) ).not.toBeTruthy();
    })

    test('debe mostrar el título en un h1', () => {

        render( <FirstApp title={title} /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML).toContain( title );
    })

    test('debe mostrar el subtítulo enviado por props', () => {

        render( <FirstApp title={title} subTitle={ subTitle } /> );

        expect( screen.getAllByAltText( subTitle).length ).toBe( 2 );

    })

    
})