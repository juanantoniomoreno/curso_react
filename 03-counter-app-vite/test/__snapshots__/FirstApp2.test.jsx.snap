// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> Debe hacer Match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Goku
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Juan
  </p>
</div>
`;

exports[`Pruebas en <FirstApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy Goku
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Juan
  </p>
</div>
`;
